File: src/com/github/serivesmejia/eocvsim/util/FpsLimiter.java
package com.github.serivesmejia.eocvsim.util;

public class FpsLimiter {

    int fps;

    double start, diff, wait;

    public FpsLimiter(int fps) {
        this.fps = fps;
    }

    public void sync() throws InterruptedException {

        wait = 1d / ((double) fps / 1000d);

        diff = System.currentTimeMillis() - start;
        if (diff < wait) {
            Thread.sleep((long) (wait - diff));
        }

        start = System.currentTimeMillis();

    }

    public void setFPS(int fps) {
        this.fps = fps;
    }

    public int getFPS() { return fps; }

}

File: EOCV-Sim/src/main/java/com/github/serivesmejia/eocvsim/util/fps/FpsLimiter.kt
package com.github.serivesmejia.eocvsim.util.fps

class FpsLimiter(var maxFPS: Int = 30) {

    var start = 0.0
    var diff = 0.0
    var wait = 0.0

    @Throws(InterruptedException::class)
    fun sync() {
        wait = 1.0 / (maxFPS.toDouble() / 1000.0)
        diff = System.currentTimeMillis() - start
        if (diff < wait) {
            Thread.sleep((wait - diff).toLong())
        }
        start = System.currentTimeMillis().toDouble()
    }

}